rxandroidble/src/main/java/com/polidea/rxandroidble/internal/operations/RxBleRadioOperationConnect.java

```diff
@@ -48,12 +50,14 @@ public Builder(
                 BleConnectionCompat connectionCompat,
                 RxBleGattCallback rxBleGattCallback,
                 @Named(CONNECT_TIMEOUT) TimeoutConfiguration connectionTimeout,
-                BluetoothGattProvider bluetoothGattProvider) {
+                BluetoothGattProvider bluetoothGattProvider,
+                Action1<RxBleConnection.RxBleConnectionState> connectionStateChangedAction) {
```

Comment:
It's not really declarative that we inject `Action1<RxBleConnection.RxBleConnectionState>` here. Can we have it wrapped in a custom interface saying that this is a connection state dispatcher? 

Looking from a test/user perspective it wouldn't say much about this parameter.